### PR TITLE
CO-41 recover provider refresh stall with free capacity

### PR DESCRIPTION
**Pull request by Kbediako**
## What
- Carry `counts.max_allowed` from `co-status --format json` into control-host supervision diagnostics.
- Keep `provider_refresh_lifecycle_stuck` restart-required health unhealthy when `running + retrying < max_allowed`, so supervision can restart/recover instead of quarantining a stuck free-capacity series.
- Refresh the CO-41 docs/task packet for the Apr 18 recurrence and add focused regression coverage for free capacity plus retry-reserved capacity.

## Why
CO-41 reopened after the April recurrence showed `stuck=true`, `restart_required=true`, `refresh_phase=refresh:claim_reconcile`, `running=2`, and `max_allowed=3`, while a Ready issue was not admitted until restart. The current blocking seam was the repeated active-worker restart quarantine treating that refresh-stuck series as healthy even though provider capacity remained open.

## Validation
- `node scripts/delegation-guard.mjs`
- `node scripts/spec-guard.mjs --dry-run`
- `npx vitest run --config vitest.config.core.ts orchestrator/tests/ControlHostSupervision.test.ts`
- `npm run build`
- `npm run lint` (existing warnings only in `orchestrator/tests/DelegationMcpHealth.test.ts`)
- `npm run test`
- `npm run docs:check`
- `npm run docs:freshness`
- `npm run repo:stewardship`
- `node scripts/diff-budget.mjs`
- `git diff --check`
- Manifest-backed review: `.runs/linear-af97d673-43a4-4a36-8738-b7f61e5b71a1/cli/2026-04-18T21-02-48-767Z-82e3a84c/review/telemetry.json` bounded success via command-intent, no actionable findings.

Linear: https://linear.app/asabeko/issue/CO-41/co-investigate-control-host-provider-refresh-stall-that-stops-new

## Summary by CodeRabbit

* **Bug Fixes**
  * Quarantine logic now respects available worker capacity, preventing unnecessary restart blocks.

* **Tests**
  * Added tests for restart-quarantine and capacity evaluation edge cases; validation run passed.

* **Documentation**
  * PRD, tech spec, action plan, task/spec docs refreshed for the Apr 18 recurrence with protected-term guidance and updated acceptance criteria.

* **Chores**
  * Task metadata, review timestamps and child‑lane approval metadata updated; removed an unnecessary API payload expansion.

### CODE REVIEW

File: tasks/specs/linear-af97d673-43a4-4a36-8738-b7f61e5b71a1.md [hunk 1]
```diff
@@ -13,40 +13,94 @@ related_tasks:
...
+- `This is only CO-211.` No: `CO-211` covers repeated restart churn with healthy active workers; `CO-41` remains the lifecycle-stall lineage anchor for this recurrence.
+- `This is only CO-214.` No: `CO-214` covers stale merged-closeout residue after `CO-211` / `PR #506`.
+- `This is only CO-248.` No: `CO-248` covers a `refresh:rehydrated` starvation shape; Apr 18 `CO-41` evidence is `refresh:claim_reconcile`.
+- `Docs refresh alone closes CO-41.` No: source repair and validation are required.
+
+## Current / Reference / Target Parity Matrix
+| Contract | Reference: March `PR #324` | Current: Apr 18 recurrence | Target |
+| --- | --- | --- | --- |
+| Issue scope | Original `CO-41` watchdog repair for stuck refresh reporting. | Reopened `CO-41` recurrence with free capacity and repeated restart quarantine. | Fix the current supervision seam without erasing March history. |
+| Refresh phase | Stale live-instance / watchdog lineage. | `refresh_phase=refresh:claim_reconcile`. | Claim-reconcile progress is covered by focused evidence and remains visible in diagnostics. |
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply tasks/specs/linear-af97d673-43a4-4a36-8738-b7f61e5b71a1.md:76]: @coderabbitai Fixed the MD058 table spacing in `tasks/specs/linear-af97d673-43a4-4a36-8738-b7f61e5b71a1.md` in commit 0484cd803. I also changed the CO-41 task evidence paths from host-local `/Users/...` form to repo-relative `.runs/...` paths across the task mirrors and docs. Scoped validation is green: `git diff --check`, `npm run docs:check`, `npm run docs:freshness`, and `node scripts/diff-budget.mjs`.
